pkg/cache/tas_flavor_snapshot.go

```diff
@@ -244,6 +245,66 @@ func (s *TASFlavorSnapshot) removeTASUsage(domainID utiltas.TopologyDomainID, us
 	s.leaves[domainID].tasUsage.Sub(usage)
 }
 
+func (s *TASFlavorSnapshot) freeCapacityPerDomain() map[string]resources.Requests {
+	freeCapacityPerDomain := make(map[string]resources.Requests)
+
+	for domainID, leaf := range s.leaves {
+		freeCapacityPerDomain[string(domainID)] = leaf.freeCapacity.Clone()
+	}
+
+	return freeCapacityPerDomain
+}
+
+func (s *TASFlavorSnapshot) tasUsagePerDomain() map[string]resources.Requests {
+	tasUsagePerDomain := make(map[string]resources.Requests)
```

Comment:
```suggestion
	tasUsagePerDomain := make(map[utiltas.TopologyDomainID]resources.Requests, len(s.leaves))
```

Using dedicated typed would be better.